src/tree-view/types.js

```diff
@@ -27,13 +27,13 @@ export type TreeViewOverridesT = {
 
 export type TreeNodeIdT = number | string;
 
-export type TreeNodeT = {
+// eslint-disable-next-line flowtype/no-weak-types
+export type TreeNodeT<T: any> = {
```

Comment:
Default generics in flow use <T = any> maybe that helps? Colon only constrains to a subtype specified. I'm not running the change locally, so not sure if this resolves the flow check